### DISCUSSION TITLE
Hope to provide the function of turning off the display of flags.

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#24369] [Ideas] [answered]
The current version of Lazer will display flags on the leaderboard. If Lazer is used in China, there is a risk of streaming accounts being banned during live streaming or recording replay.

**Comment by peppy** [comment, marked as answer]
Sorry but we can't support this. Your best bet may be to replace the game resources with transparent images or cover the flags with an overlay element.

**Comment by ghost** [comment]
> Sorry but we can't support this. Your best bet may be to replace the game resources with transparent images or cover the flags with an overlay element.

So lazer is still a risky game, which I believe is a sensitive political issue. Thank you for your reply.

**Reply by alexpado** [reply]
If you really wish to hide flags in leaderboard, either you apply Peppy solution or change `DrawableFlag.cs`

```cs
        private void load(TextureStore ts)
        {
            ArgumentNullException.ThrowIfNull(ts);
            Texture = ts.Get($@"Flags/__");
        }
```

This will force loading the "?" flag on leaderboards which is, imo, less painful than modifying every flags in the game resources.

**Reply by ghost** [reply]
It can be achieved by other means, but more importantly, I want to see the official perspective towards this matter, now I know offical perspective. This is a political issue, and it may be difficult for people outside the Chinese Mainland to understand this. I don't really want this game to have this problem, so I put it forward. Thank you for your reply.

**Reply by alexpado** [reply]
I understand your concern. One possibility could be that someone brave enough fork the osu-resources repository and modify all the flags as peppy suggested.

Conflict could be possible but likely not going to happen often as I don't think the osu! team will take fun in modifying flag every now and then. So through automation, updating that fork and compiling it into a DLL file that can be placed in osu! directory without technical knowledge is possible. One issue though is that I don't know if the game would react well to that regarding anti-cheat/online feature.

The only downside is that people might not know that such alternative exists as github is clearly not the first option to look for when you are not a developer.


To be fair the community could get on that after lazer become the main version for everyone, I can already see a lot of forum post linking game modification similar to this.

**Reply by peppy** [reply]
I think the only implementation I could get behind would be a "streaming" mode, where things like username and flags could be optionally hidden. But I'd need to see a high demand for this before implementing it.

**Reply by ghost** [reply]
Yea, anti cheat is a problem, after all, it requires modifying the game code and recompiling, so I mean there is no difficulty in the means, but if the official is willing to support this, it would be even better.

**Reply by Stehsaer** [reply]
> I think the only implementation I could get behind would be a "streaming" mode, where things like username and flags could be optionally hidden. But I'd need to see a high demand for this before implementing it.

Basically all China Mainland players who streams will need this feature. And for the community, it's also a way to avoid getting involved in present and future political issues and maintain its accessibility to all players around the globe.

**Reply by OwenCMYK** [reply]
I would be in support of implementing a streamer mode. In addition to the previously mentioned concerns, I think VTubers might want to hide their nationality under certain circumstances

**Reply by Stehsaer** [reply]
## Streaming mode features and benefits:
- **Hide Usernames or Region Flags**
  - Avoid violating laws in streamers' region.
  - Meet the needs of some players/streamers to stay anonymous.
- **Hide public chat**
  - Avoid being censored because of random inappropriate chat contents popping up in chats. osu! can, of course, ban those inappropriate according to moderator reports afterwards, but that would be too late for a streamer. Some streaming platforms have real-time monitoring over text and visual contents and will ban the streamer in a sec if any inappropriate content comes into view. 
  - Some streaming platforms have even stricter policies that prohibit players to show ANY chat area. Without streaming mode, a streamer simply can't stream on those platforms. 
  - Some online games such as APEX have already implemented this to avoid their streamers from being banned, hiding chat areas and replacing usernames with neutral replacers.
- **Hide NSFW songs/graphics**: Avoid censoring by the streaming platform, as stated before.
- **Ability for marking and hiding specific songs chosen or marked by the player**: Meeting copyright concerns and restrictions, eg. Twitch
- **In-game chat display**(optional): Extremely useful streaming feature. An in-game browser can basically do the thing.
### We really hope that the dev team can take this concern into serious consideration, since this is highly related to issue about LAW and POLITICS, and it can also PROTECT community members.
